libs/datamodel/core/src/configuration/configuration_struct.rs

```diff
@@ -29,6 +29,13 @@ impl Configuration {
         self.datasources.first().map(|source| source.referential_integrity())
     }
 
+    pub fn max_identifier_length(&self) -> usize {
+        self.datasources
+            .first()
+            .map(|source| source.active_connector.max_identifier_length())
+            .unwrap_or(usize::MAX)
+    }
```

Comment:
For this one, if I remember the API correctly, there is `configuration.active_connector()` or `datasource.active_connector()`, which is always defined and defaults to `EmptyConnector`. I don't know if the worry is founded, but `configuration.active_connector().max_identifier_length()` and `configuration.max_identifier_length()` could then be different.